Library/Homebrew/cmd/upgrade.rb

```diff
@@ -23,6 +23,14 @@ def upgrade
 
     Homebrew.perform_preinstall_checks
 
+    if ARGV.include?("--all")
+      opoo <<-EOS.undent
+        We decided to not change the behaviour of `brew upgrade` so
```

Comment:
This could probably be compressed into a simple:
```ruby
opoo "`--all` on `brew upgrade` is no longer necessary."
```
but if you're happy with the longer explanation, 👍.